### PR TITLE
fix(ci): update search path for release assets

### LINKED ISSUE #86: No assets are uploaded to the GitHub release
Unfortunately not everything worked in the 0.7.1 release. The assets were not uploaded. Not a major issue but it should be fixed for the next release.

## Logs
In case the Azure logs get too old and will be deleted.

```
2020-08-18T15:58:47.2723873Z ##[section]Starting: GitHub Upload
2020-08-18T15:58:47.2738850Z ==============================================================================
2020-08-18T15:58:47.2739452Z Task         : GitHub Release
2020-08-18T15:58:47.2739881Z Description  : Create, edit, or delete a GitHub release
2020-08-18T15:58:47.2740288Z Version      : 0.160.5
2020-08-18T15:58:47.2740640Z Author       : Microsoft Corporation
2020-08-18T15:58:47.2741691Z Help         : https://aka.ms/AA5vv5o
2020-08-18T15:58:47.2742474Z ==============================================================================
2020-08-18T15:58:47.4668852Z 9bde9970-f931-4741-b0e3-f9f2bbcbf4ae exists true
2020-08-18T15:58:47.5062106Z Computing changes made in this release...
2020-08-18T15:58:47.7352006Z Fetching the latest published release...
2020-08-18T15:58:47.9899384Z Found the latest published release: https://github.com/jellyfin/jellyfin-apiclient-java/releases/tag/v0.7.1
2020-08-18T15:58:48.2483200Z Fetching the list of commits since the last published release...
2020-08-18T15:58:48.5968714Z No changes were found. The provided target commit is the same as the commit of the last published release.
2020-08-18T15:58:48.5972148Z Release notes file: /home/vsts/work/1/s is a directory and not a file.
2020-08-18T15:58:48.5973273Z Fetching the release for tag: v0.7.1
2020-08-18T15:58:48.9265051Z Found a release for tag: v0.7.1
2020-08-18T15:58:48.9272950Z Editing the release with tag: v0.7.1
2020-08-18T15:58:49.2010463Z Uploading assets...
2020-08-18T15:58:49.2016049Z Searching for file(s) matching '**/distributions/*-${JELLYFIN_VERSION}.zip'.
2020-08-18T15:58:49.3337046Z No files found matching '**/distributions/*-${JELLYFIN_VERSION}.zip'. Nothing to upload.
2020-08-18T15:58:49.3338629Z Searching for file(s) matching '**/libs/*-${JELLYFIN_VERSION}.jar'.
2020-08-18T15:58:49.4329634Z No files found matching '**/libs/*-${JELLYFIN_VERSION}.jar'. Nothing to upload.
2020-08-18T15:58:49.4331135Z Searching for file(s) matching '**/libs/*-${JELLYFIN_VERSION}-sources.jar'.
2020-08-18T15:58:49.5278786Z No files found matching '**/libs/*-${JELLYFIN_VERSION}-sources.jar'. Nothing to upload.
2020-08-18T15:58:49.5280259Z All assets uploaded successfully.
2020-08-18T15:58:49.5281914Z Release edited successfully https://github.com/jellyfin/jellyfin-apiclient-java/releases/tag/v0.7.1
2020-08-18T15:58:49.5322867Z ##[section]Finishing: GitHub Upload
```

# Logs (after #88)
```
2020-08-19T07:33:23.5116508Z ##[section]Starting: GitHub Upload
2020-08-19T07:33:23.5129081Z ==============================================================================
2020-08-19T07:33:23.5129845Z Task         : GitHub Release
2020-08-19T07:33:23.5130467Z Description  : Create, edit, or delete a GitHub release
2020-08-19T07:33:23.5130888Z Version      : 0.160.5
2020-08-19T07:33:23.5131588Z Author       : Microsoft Corporation
2020-08-19T07:33:23.5132025Z Help         : https://aka.ms/AA5vv5o
2020-08-19T07:33:23.5133224Z ==============================================================================
2020-08-19T07:33:23.7313368Z 9bde9970-f931-4741-b0e3-f9f2bbcbf4ae exists true
2020-08-19T07:33:23.7802548Z Computing changes made in this release...
2020-08-19T07:33:23.9567074Z Fetching the latest published release...
2020-08-19T07:33:24.1384291Z Found the latest published release: https://github.com/jellyfin/jellyfin-apiclient-java/releases/tag/v0.7.2
2020-08-19T07:33:24.3440201Z Fetching the list of commits since the last published release...
2020-08-19T07:33:24.6088660Z No changes were found. The provided target commit is the same as the commit of the last published release.
2020-08-19T07:33:24.6091549Z Release notes file: /home/vsts/work/1/s is a directory and not a file.
2020-08-19T07:33:24.6101088Z Fetching the release for tag: v0.7.2
2020-08-19T07:33:24.8060860Z Found a release for tag: v0.7.2
2020-08-19T07:33:24.8061699Z Editing the release with tag: v0.7.2
2020-08-19T07:33:25.0480932Z Uploading assets...
2020-08-19T07:33:25.0488543Z Searching for file(s) matching '/home/vsts/work/1/s/**/distributions/*-${JELLYFIN_VERSION}.zip'.
2020-08-19T07:33:25.1717647Z No files found matching '/home/vsts/work/1/s/**/distributions/*-${JELLYFIN_VERSION}.zip'. Nothing to upload.
2020-08-19T07:33:25.1719290Z Searching for file(s) matching '/home/vsts/work/1/s/**/libs/*-${JELLYFIN_VERSION}.jar'.
2020-08-19T07:33:25.2674781Z No files found matching '/home/vsts/work/1/s/**/libs/*-${JELLYFIN_VERSION}.jar'. Nothing to upload.
2020-08-19T07:33:25.2675919Z Searching for file(s) matching '/home/vsts/work/1/s/**/libs/*-${JELLYFIN_VERSION}-sources.jar'.
2020-08-19T07:33:25.3548400Z No files found matching '/home/vsts/work/1/s/**/libs/*-${JELLYFIN_VERSION}-sources.jar'. Nothing to upload.
2020-08-19T07:33:25.3549354Z All assets uploaded successfully.
2020-08-19T07:33:25.3550407Z Release edited successfully https://github.com/jellyfin/jellyfin-apiclient-java/releases/tag/v0.7.2
2020-08-19T07:33:25.3588737Z ##[section]Finishing: GitHub Upload
```

### PR DESCRIPTION
**Pull request by h1dden-da3m0n**
Okay, I have now poked around in the Azure DevOps/Pipelines documentation a bit and it seems to me that they automatically move all build assets to a different path than the actual source files (eg. not `/home/vsts/work/1/s/` but to `/home/vsts/work/1/a/`). 
The only alternative that I have found that could be tried if this path fails to is `Pipeline.Workspace` (ie. `/home/vsts/work/1`) which should be the last resort to try IMHO. 

* [Build variables (DevOps Services)](https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml#build-variables-devops-services) see `Build.ArtifactStagingDirectory` description
* [Pipeline variables (DevOps Services)](https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml#pipeline-variables-devops-services) see `Pipeline.Workspace` description

:crossed_fingers: this PR works as intended and we don't have to resort to the pipeline workspace.

**Changes**

* moves azure-pipeline.yml to `.ci` folder to align with other repos
* updates JDK version used in GradleW tasks to JDK 11
* hopefully fixes search path for release assets upload
* add short readme to note the azure-pipeline schema to validate (if seen obsolete i shall remove that commit :wink: )

**Issues**

* hopefully fixes #86 once and for all

### CODE REVIEW

**Review by Maxr1998 — APPROVED** [review]

File: .ci/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+## Contribution/Development Note
+
+Azure pipeline YAMLs can be validated via a Json schema.
+The Azure DevOps documentation mentions this but sends you to get the project scoped schema, which requires authorization.
+A generic schema is although also available and can be helpful to validate:
+
+https://raw.githubusercontent.com/microsoft/azure-pipelines-vscode/master/service-schema.json
```
Comment (Maxr1998) [inline .ci/README.md:7]:
```suggestion
[A generic schema](https://raw.githubusercontent.com/microsoft/azure-pipelines-vscode/master/service-schema.json) is also available however and can be helpful for validation.
```

That should make the bot happy and tweaks the wording a bit.

File: .ci/README.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+## Contribution/Development Note
+
+Azure pipeline YAMLs can be validated via a Json schema.
+The Azure DevOps documentation mentions this but sends you to get the project scoped schema, which requires authorization.
```
Comment (Maxr1998) [inline .ci/README.md:4]:
```suggestion
The Azure DevOps documentation mentions this but asks you to get the project scoped schema, which requires authorization.
```

**Review by Maxr1998 — DISMISSED** [review]

**Comment by Maxr1998** [comment]
Oops, misclicked. Meant request changes.. :eyes:

**Review by Maxr1998 — APPROVED** [review]

File: .ci/README.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+## Contribution/Development Note
+
+Azure pipeline YAMLs can be validated via a Json schema.
```
Comment (nielsvanvelzen) [inline .ci/README.md:3]:
```suggestion
Azure Pipelines YAML files can be validated via a JSON Schema.
```
Both Azure Pipelines and JSON Schema are names so we should use the official capitalization.

File: .ci/azure-pipelines.yml [hunk 4]
```diff
@@ -71,19 +73,20 @@ jobs:
           publishJUnitResults: false
           testResultsFiles: '**/TEST-*.xml'
           javaHomeOption: 'JDKVersion'
+          jdkVersionOption: 1.11
           sonarQubeRunAnalysis: false
         env:
           BINTRAY_KEY: $(bintray_key)
 
-      - task: GithubRelease@0
+      - task: GitHubRelease@1
```
Comment (nielsvanvelzen) [inline .ci/azure-pipelines.yml:81]: I can't find a changelog for this task but I'd like to know what the changes between 0 and 1 are to see if it doesn't contain breaking changes.

**Review by nielsvanvelzen — CHANGES_REQUESTED** [review]

File: .ci/README.md (hunk 3 above)
Comment (h1dden-da3m0n) [reply .ci/README.md:3]: Sorry, will change ASAP.

File: .ci/azure-pipelines.yml (hunk 4 above)
Comment (h1dden-da3m0n) [reply .ci/azure-pipelines.yml:81]: TBH I was kind of surprised finding this too, but both v0 and v1 seem to be maintained.
There is sadly no changelog per say but here is the PR introducing [v1](https://github.com/microsoft/azure-pipelines-tasks/pull/11134)

Comment (nielsvanvelzen) [reply .ci/azure-pipelines.yml:81]: Should be no issue to update it when I read that PR. Weird how they maintain 2 separate tasks though.

**Review by Maxr1998 — APPROVED** [review]

File: .ci/azure-pipelines.yml (hunk 4 above)
Comment (h1dden-da3m0n) [reply .ci/azure-pipelines.yml:81]: I know, but it seemed save enough + no idea how long v0 would continue maintenance

**Review by nielsvanvelzen — APPROVED** [review]
Let's see if it works on the next release :)

**Pull request merged**